presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1156,14 +1159,15 @@ protected Scope visitValues(Values node, Scope scope)
 
         private void analyzeWindowFunctions(QuerySpecification node, List<Expression> outputExpressions, List<Expression> orderByExpressions)
         {
-            WindowFunctionExtractor extractor = new WindowFunctionExtractor();
-
-            for (Expression expression : Iterables.concat(outputExpressions, orderByExpressions)) {
-                extractor.process(expression, null);
+            Iterable<Expression> expressions = Iterables.concat(outputExpressions, orderByExpressions);
+            for (Expression expression : expressions) {
                 new WindowFunctionValidator().process(expression, analysis);
             }
 
-            List<FunctionCall> windowFunctions = extractor.getWindowFunctions();
+            List<FunctionCall> windowFunctions = ExpressionTreeFilter.extract(windowExtractorFunction, Lists.newArrayList(expressions))
+                    .stream()
+                    .map(FunctionCall.class::cast)
```

Comment:
See my comment (#3): https://github.com/prestodb/presto/pull/6704/commits/eea966d08823c881ed9f28b55615e723f24da292#r106887672. If you apply it then you wouldn't need to do cast here.